cmd/oras/internal/display/metadata/model/discover.go

```diff
@@ -15,21 +15,53 @@ limitations under the License.
 
 package model
 
-import ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+import (
+	"fmt"
 
-type discover struct {
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Discover is a model for discovered referrers.
+type Discover struct {
+	name  string
+	nodes map[digest.Digest]*Node
+	Root  *Node
+}
+
+// Node represents a node in the discovered reference tree.
+type Node struct {
 	Descriptor
-	Referrers []Descriptor `json:"referrers"`
+	Referrers []*Node `json:"referrers,omitempty"`
+}
+
+// Add adds a node to the discovered referrers tree.
+func (d *Discover) Add(referrer, subject ocispec.Descriptor) error {
```

Comment:
nit: Would it be clearer to name this function `AddReferrer`?

Reply:
Renamed.